proton-ge-bin: Automate switching to new GE-Proton version after update

boogiewoogit opened the pull request:
The purpose of this package is to automate the downloading/updating of GE-Proton. Since steam only sees the most recent version, every update resets the places where this GE-Proton was selected and we have to reselect the new version manually in the steam UI.

Inspired by https://aur.archlinux.org/packages/proton-ge-custom-bin, this PR changes the version as steam sees it to just be "GE-Proton", so after setting it once in the steam UI, it always points to the most recent version and doesn't have to be changed manually after updates anymore.

boogiewoogit committed:
proton-ge-bin: Automate switching to new GE-Proton version after update

proton-ge-bin: Automate switching to new GE-Proton version after update

This changes the version as steam sees it to just be "GE-Proton", so after setting it once in the steam UI, it always points to the most recent version and doesn't have to be changed manually after updates anymore.

Gliczy commented on pkgs/by-name/pr/proton-ge-bin/package.nix:
```diff
@@ -25,7 +25,14 @@ stdenvNoCC.mkDerivation (finalAttrs: {
+
+    sed -i -r 's|GE-Proton[0-9]*-[0-9]*|GE-Proton|' $steamcompattool/compatibilitytool.vdf
+    sed -i -r 's|GE-Proton[0-9]*-[0-9]*|GE-Proton|' $steamcompattool/proton
+    sed -i -r 's|GE-Proton[0-9]*-[0-9]*|GE-Proton|' $steamcompattool/version
```
I tested modifying the version in `$steamcompattool/version`, and it doesn't seem to invalidate the currently selected proton version.

So this line seems unnecessary; moreover, MangoHud can display the currently active proton version by reading it from `$steamcompattool/version`. With this line it would still indicate that proton-ge is used but not it's version.

boogiewoogit replied:
I think that merits keeping the full version information in `$steamcompattool/version`, as I have only added the modification to it to preserve internal consistency.

boogiewoogit committed:
proton-ge-bin: keep version information in `$steamcompattool/version`

proton-ge-bin: keep version information in `$steamcompattool/version`

Having full version information in `$steamcompattool/version` doesn't conflict with keeping the up-to-date GE-Proton prefix constant automatically, and it can still provide the info to analysis tools like mangoHUD.

Gliczy reviewed (approved):
LGTM. Thanks for this PR!

Titaniumtown reviewed (approved):
I love this change! Good work.
